packages/mui-codemod/src/v5.0.0/base-hook-imports.js

```diff
@@ -0,0 +1,113 @@
+const capitalize = (str) => {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+};
+
+const list = [
+  // Slider
+  'Mark',
+  // Select
+  'SelectOption',
+  'SelectChild',
+  'isOptionGroup',
+  'SelectOptionGroup',
+  'SelectChangeEventType',
+  // Listbox
+  'OptionState',
+  'UseListboxPropsWithDefaults',
+  'FocusManagementType',
+  'ListboxState',
+  'ListboxAction',
+  'ListboxReducer',
+  'ListboxReducerAction',
+  // Autocomplete
+  'createFilterOptions',
+  'FilterOptionsState',
+  'AutocompleteFreeSoloValueMapping',
+  'AutocompleteValue',
+  'UseAutocompleteProps',
+  'AutocompleteGroupedOption',
+  'CreateFilterOptionsConfig',
+  'FilterOptionsState',
+  'AutocompleteInputChangeReason',
+  'AutocompleteCloseReason',
+  'AutocompleteGetTagProps',
+];
+
+/**
+ * @param {import('jscodeshift').FileInfo} file
+ * @param {import('jscodeshift').API} api
+ */
+export default function transformer(file, api, options) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+  const printOptions = options.printOptions || { quote: 'single' };
+
+  root
+    .find(j.ImportDeclaration)
+    .filter((path) => {
+      const sourceVal = path.node.source.value;
+      return sourceVal.startsWith('@mui/base/') && sourceVal.match('Unstyled');
+    }) // Process only Base components
+    .forEach((path) => {
+      // scenario 1: `@mui/base/<Component>Unstyled/use<Component>`
```

Comment:
I don't think that this import was even supported at all. But let's keep it just in case 🤷